File: db4oj/core/src/com/db4o/MTaDontDelete.java
/* Copyright (C) 2004   db4objects Inc.   http://www.db4o.com */

package com.db4o;

import com.db4o.foundation.network.YapSocket;


/**
 * 
 */
class MTaDontDelete extends MsgD {
	final boolean processMessageAtServer(YapSocket in) {
        int classID = _payLoad.readInt();
	    int id = _payLoad.readInt();
	    Transaction trans = getTransaction();
	    YapStream stream = trans.stream();
	    synchronized (stream.i_lock) {
	        trans.dontDelete(classID, id);
	        return true;
	    }
	}
}

File: db4on/core/src/core/com/db4o/cs/messages/MTaDontDelete.cs
namespace com.db4o.cs.messages
{
	public class MTaDontDelete : com.db4o.cs.messages.MsgD
	{
		public sealed override bool ProcessMessageAtServer(com.db4o.foundation.network.YapSocket
			 @in)
		{
			int classID = _payLoad.ReadInt();
			int id = _payLoad.ReadInt();
			com.db4o.Transaction trans = GetTransaction();
			com.db4o.YapStream stream = trans.Stream();
			lock (stream.i_lock)
			{
				trans.DontDelete(classID, id);
				return true;
			}
		}
	}
}
